### DISCUSSION TITLE
Backwards compatability between 1.7.1 and 2.x

### DISCUSSION THREAD

**Discussion by jpgreenfield** [resilience4j/resilience4j#1717] [Q&A] [answered]
Hi there,

We are looking to use resilience4j as a retrying framework. So far, all looks good in my local use/testing. However, I've been asked about Java 11+ support and the fact that the 1.7.1 release was about 1 year back.

So, there's some concern in our team over newer updates being available and things being maintained. I saw the other discussion about 2.x work for Java 11, so that sounds good. But, what I wanted to confirm is whether the code we write now with 1.7.1 will still work with the newer 2.x versions in general (i.e. backwards compatibility for existing features)? I'm a bit worried if we have the retrying working now, whether moving to 2.x later will require us and/or our users to have a lot of re-work. Could you please provide some insight into this?

Thank you,

--Jason

**Comment by RobWin** [comment, marked as answer]
Hi,
2.x won't be backwards-compatible, because we removed Vavr as a dependency and switched to Java11.
But besides package migrations there shouldn't be much effort. The API itself hasn't changed.
